### PR TITLE
[Keyboard] Add new keyboard soda/mango

### PR DESCRIPTION
**Pull request by gezhaoyou**
## Description

 Add new keyboard soda/mango
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/soda/mango/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    rgb_matrix_set_color_all(0, 0, 0);
+    rgb_matrix_set_suspend_state(true);
+}
+
+void suspend_wakeup_init_user(void) {
+    // code will run on keyboard wakeup
+    rgb_matrix_set_suspend_state(false);
+}
+
+
```
Comment (zvecr) [inline keyboards/soda/mango/keymaps/default/keymap.c:49]:
```suggestion
```

File: keyboards/soda/mango/chconf.h
Comment (zvecr) [inline keyboards/soda/mango/chconf.h:1]: This file should be removed

**Review by zvecr — CHANGES_REQUESTED** [review]
From <https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs>

```
info.json
With the move to [data driven](https://docs.qmk.fm/#/data_driven_config) keyboard configuration, we encourage contributors to utilise as many features as possible of the info.json [schema](https://github.com/qmk/qmk_firmware/blob/master/data/schemas/keyboard.jsonschema).
the mandatory elements for a minimally complete info.json at present are:
valid URL
valid maintainer
valid USB VID/PID and device version
displays correctly in Configurator (press Ctrl+Shift+I to preview local file, turn on fast input to verify ordering)
layout definitions should include matrix positions, so that LAYOUT macros can be generated at build time
should use standard definitions if applicable
use the Community Layout macro names where they apply (preferred above LAYOUT/LAYOUT_all)
If the keyboard only has a single electrical/switch layout:
use LAYOUT as your macro name, unless a community layout already exists
If the keyboard has multiple electrical/switch layouts:
include a LAYOUT_all which specifies all possible layout positions in the electrical matrix
use alternate layout names for all other possible layouts, preferring community layout names if an equivalent is available (e.g. LAYOUT_tkl_ansi, LAYOUT_ortho_4x4 etc.)
Microcontroller and bootloader
Diode Direction (if not using direct pins)
the following are required to be configured in info.json if necessary
Direct pin configuration
Backlight Configuration (where applicable)
Split keyboard configuration (where applicable)
Encoder Configuration
Bootmagic Configuration
LED Indicator Configuration
```

File: keyboards/soda/mango/config.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+#pragma once
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+// enable NKRO_ENABLE
+#define FORCE_NKRO
+// The pin connected to the data pin of the LEDs
+#define RGB_DI_PIN A2
```
Comment (drashna) [inline keyboards/soda/mango/config.h:25]: This is changed in develop, and lands on master in a week.  To ensure that it compiles: 
```suggestion
#define RGB_DI_PIN A2
#define WS2812_DI_PIN A2
```

File: keyboards/soda/mango/mango.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "mango.h"
+
+#ifdef RGB_MATRIX_ENABLE
+
+// RGB Matrix 
+led_config_t g_led_config = {
```
Comment (drashna) [inline keyboards/soda/mango/mango.c:23]: This should be moved to the info.json config.  You can run `qmk info -k soda/mango -f json` for assistance with that.

File: keyboards/soda/mango/mango.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
+
+#define ___ KC_NO
+
+#define LAYOUT_numpad_5x4( \
```
Comment (drashna) [inline keyboards/soda/mango/mango.h:24]: As above -> info.json.

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**